packages/web3-net/types/index.d.ts

```diff
@@ -0,0 +1,22 @@
+/*
+    This file is part of web3.js.
+    web3.js is free software: you can redistribute it and/or modify
+    it under the terms of the GNU Lesser General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+    web3.js is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU Lesser General Public License for more details.
+    You should have received a copy of the GNU Lesser General Public License
+    along with web3.js.  If not, see <http://www.gnu.org/licenses/>.
+*/
+/**
+ * @file index.d.ts
+ * @author Josh Stevens <joshstevens19@hotmail.co.uk>
+ * @date 2018
+ */
+
+import { NetworkBase } from 'web3-core';
+
+export class Network extends NetworkBase {}
```

Comment:
isn't it possible to export it directly with the renaming of NetworkBase to Network?

Reply:
don#t think that's possible in TS, hence why i went down the route of having base classes which match the network classes and can be shared. You can't have the same names of imported classes and exported classes so i could not call it `Network` in a TS world it doesn't matter and web3-core is not a public interface so they would never be importing `NetworkBase`. Even if they did then it would match the `Network` class anyway :).